pkg/gofr/http/middleware/metrics.go

```diff
@@ -21,11 +22,19 @@ type metrics interface {
 func Metrics(metrics metrics) func(inner http.Handler) http.Handler {
 	return func(inner http.Handler) http.Handler {
 		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			// Compile regex for common file extensions
+			staticFileRegex := regexp.MustCompile(`\.(css|js|png|jpg|jpeg|gif|ico|svg|txt|html|json|woff|woff2|ttf|eot|pdf)$`)
```

Comment:
also, if you want to keep a regex, move it away from the method to a var in package level

Otherwise, you will compute the regex on each metrics call